### PR TITLE
[Packetbeat] Fix SIP parser logic related to line length check

### LINKED ISSUE #23410: [PacketBeat][SIP] Error in parsing of 200 OK
- Version: 7.10 and master
- Operating System: any
- Steps to Reproduce: Try and parse a 200 OK message, get error

Hi.

We did not get any 200 OK messages passed from packetbeat to elastic, so we started investigating it.
This is an example of a log message from packetbeat:
2021-01-08T14:53:31.689+0100        DEBUG        [udp]        udp/udp.go:74        Parsing packet from IpPortTuple src[xxxx:5060] dst[yyyy:5060] of length 871.
2021-01-08T14:53:31.689+0100        DEBUG        [sipdetailed]        sip/plugin.go:112        Payload received: [SIP/2.0 200 OK
Via: SIP/2.0/UDP xxxx:5060;rport=5060;received=xxxx;branch=z9hG4bK08B5783418fa90e0933
Call-ID: 07d27b45-2622-4df2-aea9-2b1a02a24d8b
From: <sip:+aaaa@xxxx>;tag=gK08dae5b1
To: <sip:+bbbb@yyyy>;tag=e36e99a2-4f00-4112-ae7b-767a84e9e5eb
CSeq: 776 INVITE
Session-Expires: 1800;refresher=uas
Contact: <sip:ffff@yyyy:5060>
Allow: OPTIONS, REGISTER, SUBSCRIBE, NOTIFY, PUBLISH, INVITE, ACK, BYE, CANCEL, UPDATE, PRACK, MESSAGE, REFER
Supported: 100rel, timer, replaces, norefersub
Server: eeee
Content-Type: application/sdp
Content-Length:   232
[1B blob data]
v=0
o=- 674143628 674143630 IN IP4 xxxx
s=eeee
c=IN IP4 xxxx
t=0 0
m=audio 23464 RTP/AVP 8 97
a=rtpmap:8 PCMA/8000
a=rtpmap:97 telephone-event/8000
a=fmtp:97 0-16
a=ptime:20
a=maxptime:150
a=sendrecv
]
2021-01-08T14:53:31.689+0100        DEBUG        [sip]        sip/parser.go:167        First line too small
2021-01-08T14:53:31.689+0100        DEBUG        [sip]        sip/plugin.go:106        error: first line too small

Request has been verified to have CRLF in pcap.

This is the code:
func (*parser) parseSIPLine(pi *parsingInfo, m *message) error {
	// ignore any CRLF appearing before the start-line (RFC3261 7.5)
	pi.data = bytes.TrimLeft(pi.data[pi.parseOffset:], string(constCRLF))

	i := bytes.Index(pi.data[pi.parseOffset:], constCRLF)
	if i == -1 {
		return errors.New("not found expected CRLF")
	}

	// Very basic tests on the first line. Just to check that
	// we have what looks as a SIP message
	var (
		version []byte
		err     error
	)

	fline := pi.data[pi.parseOffset:i]
	if len(fline) < 16 { // minimum line will be "SIP/2.0 XXX OK\r\n"
		if isDebug {
			debugf("First line too small")
		}
		return errors.New("first line too small")
	}

Running it in our head (and assuming parseOffset is 0 or whatever start index Go uses) we get

SIP/2.0 200 OK\r\n
	pi.data = bytes.TrimLeft(pi.data[pi.parseOffset:], string(constCRLF))
SIP/2.0 200 OK\r\n
	i := bytes.Index(pi.data[pi.parseOffset:], constCRLF)
i = 14
	fline := pi.data[pi.parseOffset:i]
fline = SIP/2.0 200 OK
	if len(fline) < 16 { // minimum line will be "SIP/2.0 XXX OK\r\n"
will always be true, since 14 is always lower than 16, and the comment is wrong, since the \r\n has been stripped.

A fix would be to have <14 (and fix the comment to not include \r\n) OR change fline := pi.data[pi.parseOffset:i] to fline := pi.data[pi.parseOffset:i+2]

Any thoughts?

### PR DESCRIPTION
**Pull request by ljw-**
## What does this PR do?

Changed 16 -> 14, since CRLF is already stripped

## Why is it important?

Without this no 200 OK messages will be sent to elasticsearch

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~- [ ] I have made corresponding changes to the documentation~
~- [ ] I have made corresponding change to the default configuration files~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes #23410

### CODE REVIEW

**Comment by marc-gr** [comment]
Hello @ljw- ! Thanks for the fix! 

There are some more things you need to do so we can merge it:

- Please sign the [Contributor Agreement](https://www.elastic.co/contributor-agreement) (using the same email you used to perform the commits).
- Add an entry at the end of `CHANGELOG.next.asciidoc` file under the `fixed / packetbeat` section explaining your change briefly and referencing the issue/pr.
- If possible, add some tests for the fix: you can either add a unit test or an integration one, (ideally both 😄  ) if you need help doing that, please let us know and we can help you through it, or if you prefer to give us push access to the PR we can do it ourselves, as you prefer 😃 .

Thanks!

File: packetbeat/protos/sip/parser.go [hunk 1]
```diff
@@ -165,7 +165,7 @@ func (*parser) parseSIPLine(pi *parsingInfo, m *message) error {
 	)
 
 	fline := pi.data[pi.parseOffset:i]
-	if len(fline) < 16 { // minimum line will be "SIP/2.0 XXX OK\r\n"
+	if len(fline) < 14 { // minimum line will be "SIP/2.0 XXX OK"
```
Comment (andrewkroh) [inline packetbeat/protos/sip/parser.go:168]: I think using a constant like this would be more clear.

`const minStatusLineLength = len("SIP/2.0 XXX OK")`

**Comment by ljw-** [comment]
Hello @marc-gr .

- I did sign the agreement right after making this pull request. How do you/I relate it to my Github user account?
- CHANGELOG.next.asciidoc: Added, hopefully in the right place?
- Tests, I dont really have a Go environment setup and don't know Go and testing in Go, so I'd be happy if you do it. Googling "push access to the PR" it seems "Allow edits by maintainers" needs to be checked on this PR, which it is. Is there anything else I need to do for you to be able to add the tests to the PR?

Anything else? :)

**Comment by marc-gr** [comment]
> Hello @marc-gr .
> 
> * I did sign the agreement right after making this pull request. How do you/I relate it to my Github user account?
> * CHANGELOG.next.asciidoc: Added, hopefully in the right place?
> * Tests, I dont really have a Go environment setup and don't know Go and testing in Go, so I'd be happy if you do it. Googling "push access to the PR" it seems "Allow edits by maintainers" needs to be checked on this PR, which it is. Is there anything else I need to do for you to be able to add the tests to the PR?
> 
> Anything else? :)

The CLA is signed now, thanks! I will be pushing the tests next 😄

**Review by marc-gr — APPROVED** [review]
LGTM as far as CI passes

**Comment by marc-gr** [comment]
jenkins run tests

**Pull request merged**